File: app/src/main/java/com/neomu/neomu/app/index/LoginActivity.java
package com.neomu.neomu.app.index;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.neomu.neomu.R;
import com.neomu.neomu.app.club.MainActivity;
import com.neomu.neomu.app.index.util.JoinValidation;
import com.neomu.neomu.common.activity.BaseActivity;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

//로그인 액티비티
public class LoginActivity extends BaseActivity{

    FirebaseAuth firebaseAuth;

    @BindView(R.id.login_id) EditText login_id;
    @BindView(R.id.login_pw) EditText login_pw;
    @BindView(R.id.login_btn) Button login_btn;
    @BindView(R.id.main_linkText) TextView main_linkText;
    public final String MSG_SIGNIN_FAIL = "로그인 실패";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        ButterKnife.bind(this);
        firebaseAuth = FirebaseAuth.getInstance();
    }

    @OnClick({R.id.login_btn, R.id.main_linkText})
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.login_btn:
                new JoinValidation().isValidation(login_id, null, login_pw, null, new JoinValidation.ValidationListener() {
                    @Override
                    public void onInvalid(View view, String msg) {
                    }

                    @Override
                    public void onSuccess() {
                        firebaseAuth.signInWithEmailAndPassword(login_id.getText().toString(), login_pw.getText().toString())
                                .addOnCompleteListener(LoginActivity.this, new OnCompleteListener<AuthResult>() {
                                    @Override
                                    public void onComplete(@NonNull Task<AuthResult> task) {
                                        if (task.isSuccessful()) {
                                            startActivity(new Intent(LoginActivity.this, MainActivity.class));
                                            finish();
                                        } else {
                                            showToast(MSG_SIGNIN_FAIL, true);
                                        }
                                    }
                                });
                    }
                });
                case R.id.main_linkText:
                    Intent intent = new Intent(LoginActivity.this, AuthActivity.class);
                    startActivity(intent);
        }
    }
}

File: app_kt/src/main/java/com/example/app_kt/app/main/activity/LoginActivity.kt
package neomu.kotlin.app.main

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.EditText
import com.example.app_kt.R
import com.example.app_kt.app.main.activity.AuthActivity
import neomu.kotlin.common.activity.BaseActivity
import neomu.kotlin.common.util.ValidationUtil
import neomu.kotlin.common.util.ValidationUtil.Companion.isSuccessSignInFirebaseAuth
import neomu.kotlin.common.util.ValidationUtilListener
import org.jetbrains.anko.intentFor
import org.jetbrains.anko.toast

class LoginActivity: BaseActivity(), View.OnClickListener {

    lateinit var forInvalidChecklist : MutableList<String>
    lateinit var accountTextMap : MutableMap<String, String>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
    }

    override fun onClick(v: View?) {
        when(v?.id) {
            R.id.login_btn -> {
                ValidationUtil.isValidate(forInvalidChecklist, object : ValidationUtilListener {
                    override fun onInvalid() {

                    }

                    override fun onSuccess() {
                        signInFirebaseAuth()
                    }

                })
            }
            R.id.main_linkText -> {
                startActivity(intentFor<AuthActivity>("logTest" to "onNewIntent() 실행")
                                .setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP))
            }
        }
    }

    private fun signInFirebaseAuth() {
        isSuccessSignInFirebaseAuth(this, getUserIdForLogin(), getUserPwForLogin())
    }

    private fun getAccountForLogin() : MutableMap<String, String> {
        accountTextMap = mutableMapOf("userId" to getUserIdForLogin(), "userPw" to getUserPwForLogin())
        return accountTextMap
    }

    private fun getUserIdForLogin() : String {
        val name = findViewById<EditText>(R.id.login_id)?.text.toString()
        return if (name.isNullOrEmpty()) "" else name
    }

    private fun getUserPwForLogin() : String {
        val password = findViewById<EditText>(R.id.login_pw)?.text.toString()
        return if(password.isNullOrEmpty()) "" else password
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        toast(intent?.getStringExtra("logTest").toString())
    }
}